sentry-core/src/main/java/io/sentry/core/Hub.java

```diff
@@ -18,52 +39,100 @@ public boolean isEnabled() {
 
   @Override
   public SentryId captureEvent(SentryEvent event) {
-    return null;
+    SentryId sentryId;
+    StackItem item = stack.peek();
+    sentryId = item.client.captureEvent(event, item.scope);
+    this.lastEventId = sentryId;
+    return sentryId;
   }
 
   @Override
   public SentryId captureMessage(String message) {
-    return null;
+    SentryId sentryId;
+    StackItem item = stack.peek();
+    sentryId = item.client.captureMessage(message, item.scope);
+    this.lastEventId = sentryId;
+    return sentryId;
   }
 
   @Override
   public SentryId captureException(Throwable throwable) {
-    return null;
+    SentryId sentryId;
+    StackItem item = stack.peek();
+    sentryId = item.client.captureException(throwable, item.scope);
+    this.lastEventId = sentryId;
+    return sentryId;
   }
 
   @Override
   public void close() {
```

Comment:
This impl makes sense in the light of `IHub` extending `ISentryClient` but is highly confusing otherwise :confused:

Reply:
Hub is still the object one uses to dispatch calls to the client that involve Scope.
Closing the hub just means closing the client it holds. 

`Sentry.close` holds `Hub`s. The alternative would be to expose the `SentryClient` a `Hub` holds so one can call `close`.

Why would you say it's confusing?

Reply:
Because hub holds a stack of scopes yet close is only closing the client of the top of the stack. If all scopes are supposed to reference the same client, then I think the StackItem should NOT be static as I suggested above, but should also NOT have the client as an explicit field.